src/file.ts

```diff
@@ -1494,7 +1495,16 @@ class File extends ServiceObject<File> {
         rawResponseStream.on('error', onComplete);
 
         const headers = rawResponseStream.toJSON().headers;
-        isServedCompressed = headers['content-encoding'] === 'gzip';
+        isCompressed = headers['content-encoding'] === 'gzip';
+
+        // The object is safe to validate if:
+        // 1. It was stored gzip and returned to us gzip OR
+        // 2. It was never stored as gzip
+        safeToValidate =
+          (headers['x-goog-stored-content-encoding'] === 'gzip' &&
+            isCompressed) ||
+          headers['x-goog-stored-content-encoding'] === 'identity';
+
```

Comment:
Which tests validate the following use cases?
- `headers['x-goog-stored-content-encoding'] === 'gzip' && headers['content-encoding'] === 'gzip'`
- `headers['x-goog-stored-content-encoding'] === 'gzip' && headers['content-encoding'] !== 'gzip'`
- `headers['x-goog-stored-content-encoding'] !== 'gzip' && headers['x-goog-stored-content-encoding'] === 'identity'`

Reply:
Good callout. I was looking at it more like tests for `true` `safeToValidate` and tests for `false` safeToValidate`. I'll add additional tests for the specific situations.

Reply:
@danielbankhead added tests for cases 1 & 2, case 3 is covered by existing tests